core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -803,6 +803,13 @@ package object config {
       .booleanConf
       .createWithDefault(true)
 
+  private[spark] val METRICS_EXECUTOR_SOURCE_ENABLED =
+    ConfigBuilder("spark.metrics.executorSource.enabled")
```

Comment:
Hi, @LucaCanali .
This is a completely orthogonal topic from the PR title, `Register the executor source with the metrics system when running in local mode`. Could you make a separate PR for this first?

Reply:
Technically, `spark.metrics.executorSource.enabled=false` doesn't make senses to me. Do you need to disable this?

Reply:
Thanks @dongjoon-hyun for the quick reaction on this. My reasoning for adding the possibility of turning the executor source off in this PR, was that I thought it could help adoption, in the sense that it could be a safety net for those who may be impacted by the addition of this list of new driver metrics (in local mode). Also it can be seen as a "simmetric partner" of `spark.metrics.executorMetricsSource.enabled`. I actually have no use case that needs to turn the executor source metrics off, so I can for sure split this part out.

Reply:
Can we reuse `spark.metrics.executorMetricsSource.enabled` in the `local mode`?

Reply:
`ExecutorMetricsSource` is a new one in 3.0, but `ExecutorSource` is very old one. If we don't have a use case, let's not add this configuration.

Reply:
OK.